src/Psy/Plugin/Manager.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Psy\Plugin;
+
+class Manager
+{
+    /** @var AbstractPlugin[] */
+    protected static $plugins = array();
+
+    /**
+     * @param AbstractPlugin $plugin
+     * @param $name
+     */
+    public static function register(AbstractPlugin $plugin, $name)
+    {
+        self::$plugins[$name] = $plugin;
+    }
+
+    /**
+     * @param array $configuration
+     *
+     * @return array
+     */
+    public static function getConfiguration($configuration = array())
+    {
+        foreach (self::$plugins as $plugin) {
+            $configuration = $plugin::getConfiguration($configuration);
```

Comment:
you forgot some things here. You're overwrting the `$configuration` with the last plugin found in the `$plugins` array, and overwriting the parameter `$configuration`, so it is not correct anymore with the 2nd+ plugin...